core/domain/stats_jobs_one_off.py

```diff
@@ -1275,3 +1275,59 @@ def reduce(key, stringified_values):
                     'all: %s sum:%s' % (
                         key, state_name, all_state_hit[state_name],
                         sum_state_hit[state_name]),)
+
+
+class RegenerateMissingStatsModels(jobs.BaseMapReduceOneOffJobManager):
+    """A one-off job to regenerate stats models which were missing due to
+    incorrect handling of exploration reverts. If a model is missing at version
+    x, we will regenerate all models from version x-1 till the max version of
+    the exploration.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(exp):
+        all_models = (
+            stats_models.ExplorationStatsModel.get_multi_stats_models(
+                [exp_domain.ExpVersionReference(exp.id, version)
+                 for version in range(1, exp.version + 1, 1)]))
+        first_missing_version = -1
+        for i, model in enumerate(all_models):
+            if model is None:
+                first_missing_version = i + 1
+
+        if first_missing_version == -1:
+            yield ('No change', exp.id)
+
+        for i in range(first_missing_version - 1, exp.version + 1, 1):
```

Comment:
Hm what if first_missing_version is 1? Then your next line gets the commit corresponding to version 0, which might be a problem.

Reply:
Added a check! This case (also) shouldn't happen though. Added the log anyway :)